app/supplementary_data.py

```diff
@@ -0,0 +1,89 @@
+import json
+from typing import Mapping
+
+from flask import current_app
+from marshmallow import ValidationError
+from requests import RequestException
+from structlog import get_logger
+
+from app.utilities.request_session import get_retryable_session
+from app.utilities.supplementary_data_parser import validate_supplementary_data_v1
+
+SUPPLEMENTARY_DATA_URL = "http://localhost:5003/v1/unit_data"
+SUPPLEMENTARY_DATA_REQUEST_BACKOFF_FACTOR = 0.2
+SUPPLEMENTARY_DATA_REQUEST_MAX_RETRIES = 2  # Totals no. of request should be 3. The initial request + SUPPLEMENTARY_DATA_REQUEST_MAX_RETRIES
+SUPPLEMENTARY_DATA_REQUEST_TIMEOUT = 3
+SUPPLEMENTARY_DATA_REQUEST_RETRY_STATUS_CODES = [
+    408,
+    429,
+    500,
+    502,
+    503,
+    504,
+]
+
+logger = get_logger()
+
+
+class SupplementaryDataRequestFailed(Exception):
+    def __str__(self) -> str:
+        return "Supplementary Data request failed"
+
+
+def get_supplementary_data(dataset_id: str, unit_id: str, survey_id: str) -> dict:
+    supplementary_data_url = current_app.config["SDS_API_BASE_URL"]
+
+    constructed_supplementary_data_url = (
+        f"{supplementary_data_url}?dataset_id={dataset_id}&unit_id={unit_id}"
+    )
```

Comment:
Minor, but use urlencode for safety as we don't control the ids.

```py
parameters = {
    'dataset_id': dataset_id,
    'unit_id': unit_id
}

encoded_parameters = urlencode(parameters)
constructed_supplementary_data_url = f"{supplementary_data_url}?{encoded_parameters}"
```

Reply:
Updated 👍